File: my-folder/1116-maximum-level-sum-of-a-binary-tree/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxLevelSum(TreeNode root) {
        if (root == null) {
            return 0;
        }
        Queue<TreeNode> q = new LinkedList<>();
        q.offer(root);
        int max = Integer.MIN_VALUE;
        int res = 1;
        int level = 0;
        while (!q.isEmpty()) {
            int size = q.size();
            level++;
            int total = 0;
            for (int i = 0; i < size; i++) {
                TreeNode curr = q.poll();
                total += curr.val;
                if (curr.left != null) {
                    q.offer(curr.left);
                }
                if (curr.right != null) {
                    q.offer(curr.right);
                }
            }  
            if (total > max) {
                res = level;
                max = total;
            }
        }
        return res;
    }
}

File: my-folder/1116-maximum-level-sum-of-a-binary-tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        q = deque([root])
        max_sum = float('-inf')
        res = 1
        level = 1
        while q:
            current_sum = 0
            size = len(q)
            for i in range(size):
                node = q.popleft()
                current_sum += node.val
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
            if current_sum > max_sum:
                max_sum = current_sum
                res = level
            level += 1
        return res



